tests/compiler_driver/inlinehal_executable.mlir

```diff
@@ -0,0 +1,18 @@
+// RUN: iree-compile \
+// RUN:   --mlir-print-ir-after=inline \
```

Comment:
you can use `--compile-to=hal` here to reliably get the HAL IR instead of needing to print ir after the inliner runs - you then also don't need an output
you'll also want to use vmvx as we try to keep common tests on a backend that is always available (llvm-cpu can be disabled)

Reply:
vmvx inline is not planned to be supported based on https://github.com/openxla/iree/issues/12620

I will make the other change in a new commit, thanks.